extensions/azurecore/src/azurecore.d.ts

```diff
@@ -75,10 +82,17 @@ declare module 'azurecore' {
 		getStorageAccounts(account: azdata.Account, subscriptions: azureResource.AzureResourceSubscription[], ignoreErrors?: boolean): Promise<GetStorageAccountResult>;
 		getBlobContainers(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, storageAccount: azureResource.AzureGraphResource, ignoreErrors?: boolean): Promise<GetBlobContainersResult>;
 		getFileShares(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, storageAccount: azureResource.AzureGraphResource, ignoreErrors?: boolean): Promise<GetFileSharesResult>;
-		getMigrationController(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?: boolean): Promise<GetMigrationControllerResult>;
-		createMigrationController(account:azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?:boolean): Promise<CreateMigrationControllerResult>;
-		getMigrationControllerAuthKeys(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?: boolean): Promise<GetMigrationControllerAuthKeysResult>;
-
+		/**
+		 * Makes Azure REST requests to create, retireve, update or delete access to azure service's resources.
+		 * For reference to different service URLs, See https://docs.microsoft.com/en-us/rest/api/?view=Azure
+		 * @param account The azure account used to acquire access token
+		 * @param subscription The subscription under azure account where the service will perform operations.
+		 * @param serviceUrl The endpoint for the service. See https://docs.microsoft.com/en-us/rest/api/?view=Azure
+		 * @param requestType Http request method. Currently GET, PUT, POST and DELETE methods are supported.
+		 * @param requestBody Optional request body to be used in PUT and POST requests.
+		 * @param ignoreErrors When this flag is set the method will not throw any runtime or service errors and will return the error in errors array.
+		 */
+		makeAzureRestRequest(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, serviceUrl: string, requestType: HttpRequestMethod, requestBody?: any, ignoreErrors?: boolean): Promise<AzureRestResponse>;
```

Comment:
For now I think we should control the specific endpoints by having them pass in an enum from this list https://github.com/microsoft/azuredatastudio/blob/main/extensions/azurecore/src/account-provider/providerSettings.ts#L12 and then pick the actual endpoint from that based on the current configured cloud - this way the callers don't have to know about the different cloud endpoints.

Reply:
Although you'll have to figure out what to do about the region-specific host you're using currently. I can't imagine that's going to be a common scenario so I'm hesitant to build that in for everything but I suppose better to do that now and have it even if we don't use it for most things.